subsquid/src/pabloProcessor.ts

```diff
@@ -0,0 +1,299 @@
+    console.info('processing PoolCreatedEvent', event);
+    const poolCreatedEvt = getPoolCreatedEvent(event);
+    const owner = ss58.codec("picasso").encode(poolCreatedEvt.owner);
+    const pool = await getOrCreate(ctx.store, PabloPool, poolCreatedEvt.poolId.toString());
+    // only set values if the owner was missing, i.e a new pool
+    if (pool.owner == null) {
+        let timestamp = BigInt(new Date().getTime());
+        pool.owner = owner;
+        pool.poolId = poolCreatedEvt.poolId.toString();
+        pool.quoteAssetId = poolCreatedEvt.assets.quote;
+        pool.transactionCount = 1;
+        pool.totalLiquidity = '0.0';
+        pool.totalVolume = '0.0';
+        pool.calculatedTimestamp = timestamp;
+        pool.blockNumber = BigInt(ctx.block.height);
+
+        let tx = await get(ctx.store, PabloTransaction, ctx.event.id);
+        if (tx != undefined) {
+            console.error("Unexpected transaction in db", tx);
+            throw new Error("Unexpected transaction in db");
+        }
+        tx = createTransaction(ctx, pool,
+            owner,
+            PabloTransactionType.CREATE_POOL,
+            // Following fields are irrelevant for CREATE_POOL
+            '0',
+            poolCreatedEvt.assets.base,
+            BigInt(0),
+            poolCreatedEvt.assets.quote,
+            BigInt(0));
+
+        let quoteAsset = await get(ctx.store, PabloPoolAsset, createPoolAssetId(pool.poolId, poolCreatedEvt.assets.quote));
+        let baseAsset = await get(ctx.store, PabloPoolAsset, createPoolAssetId(pool.poolId, poolCreatedEvt.assets.base));
+        if (quoteAsset != undefined || baseAsset != undefined) {
+            console.error("Unexpected assets for pool in db", quoteAsset, baseAsset);
+            throw new Error("Unexpected assets found");
+        }
+        quoteAsset = createAsset(pool, poolCreatedEvt.assets.quote, ctx, timestamp);
+        baseAsset = createAsset(pool, poolCreatedEvt.assets.base, ctx, timestamp);
+
+        await ctx.store.save(pool);
+        await ctx.store.save(baseAsset);
+        await ctx.store.save(quoteAsset);
+        await ctx.store.save(tx);
+    }
+}
+
+function createPoolAssetId(poolId: string, assetId: bigint): string {
+    return poolId + '-' + assetId;
+}
+
+interface LiquidityAddedEvent {
+    who: Uint8Array,
+    poolId: bigint,
+    baseAmount: bigint,
+    quoteAmount: bigint,
+    mintedLp: bigint
+}
+
+function getLiquidityAddedEvent(event: PabloLiquidityAddedEvent): LiquidityAddedEvent {
+    if (event.isV2100) {
+        const {who, poolId, baseAmount, quoteAmount, mintedLp} = event.asV2100;
+        return {who, poolId, baseAmount, quoteAmount, mintedLp};
+    } else {
+        const {who, poolId, baseAmount, quoteAmount, mintedLp} = event.asLatest;
+        return {who, poolId, baseAmount, quoteAmount, mintedLp};
+    }
+}
+
+export async function processLiquidityAddedEvent(ctx: EventHandlerContext, event: PabloLiquidityAddedEvent) {
```

Comment:
Reviewer note: @hussein-aitlahcen please review this part carefully for any mistakes :)